core/models/machine.py

```diff
@@ -62,20 +62,21 @@ def test_existence(cls, provider, identifier):
     def _split_cloud_name(cls, machine_name):
         version_sep = settings.APPLICATION_VERSION_SEPARATOR
         if version_sep in machine_name:
-            split_list = machine_name.split(version_sep)
+            split_list = [s.strip() for s in machine_name.split(version_sep)]
 
         if len(split_list) == 1:
             logger.warn(
                 "Version separator(%s) was not found: %s"
                 % (version_sep, machine_name))
-            split_list = [split_list[0].trim(), '']
-
-        if len(split_list) > 2:
+            split_list = [split_list[0], '']
+        elif len(split_list) > 2:
             logger.warn(
                 "Version separator(%s) is ambiguous: %s"
                 % (version_sep, machine_name))
-            version_parts = machine_name.rpartition(version_sep)
-            split_list = [version_parts[0].trim(), version_parts[2].trim()]
+            version_parts = [
+                s.strip() for s in
+                machine_name.rpartition(version_sep)]
+            split_list = [version_parts[0], version_parts[2]]
```

Comment:
These 10 or so lines of string splitting can be avoided by a dedicated version field in the metadata. Code is hard to verify and a name should not be parsed for semantic meaning (because a names purpose is to label not to define its parts (thats what data is for), all names should be valid, and a user expects this, changing the name should not break atmo, changing the metadata could)

Reply:
Agreed. Will move away from this entirely.

Reply:
Method still exists, but is no longer in use within the codebase.
```
git grep _split_cloud_name
core/models/machine.py:    def _split_cloud_name(cls, machine_name):
```